### PR TITLE
Add CLI option: `--reporter-keys`

### PR DESCRIPTION
**Pull request by jugglinmike**
The ability to specify a whitelist of keys to include in JSON output is
useful when working with large numbers of tests (where including all
data would result in very large JSON files).

### CODE REVIEW

File: bin/run.js [hunk 1]
```diff
@@ -32,13 +32,24 @@ if (argv.version) {
...
 } else {
   console.error(`Reporter ${argv.reporter} not found.`);
   process.exit(1);
 }
 
+if (argv.reporterKeys) {
+  if (argv.reporter !== 'json') {
+    console.error('`--reporter-keys` option applies only to the `json` reporter.');
+    process.exitCode = 1;
+    return;
```
Comment (rwaldron) [inline bin/run.js]: The convention used is `process.exit(1);`

Comment (jugglinmike) [inline bin/run.js]: Would you like me to fix that, first?

Comment (jugglinmike) [inline bin/run.js]: Ah, my response is a little ambiguous. Let me rephrase: would you like me to submit a separate pull request that corrects the convention prior to merging this pull request?

Comment (rwaldron) [inline bin/run.js]: Please make the change in this PR. Thank you

File: test/test.js [hunk 2]
```diff
@@ -16,26 +20,16 @@ function reportRunError(error) {
...
-        'test/collateral/bothStrict.js'
-      );
-    } else {
-      args.push('test/collateral/**/*.js');
-    }
+        '--hostType', 'node',
+        '--hostPath', process.execPath,
+        '-r', 'json',
+        '--includesDir', './test/test-includes',
+      ].concat(extraArgs);
```
Comment (rwaldron) [inline test/test.js]: 👍

**Comment by jugglinmike** [comment]
@rwaldron I've updated the convention as requested.

**Comment by bterlson** [comment]
What do you think about defaulting the list of keys to something reasonable? Most people wont' care about the compiled test string, at least...

**Comment by bterlson** [comment]
Ahh, I see. ;)
